clients/src/main/java/org/apache/kafka/clients/InFlightRequests.java

```diff
@@ -90,9 +90,7 @@ public void add(NetworkClient.InFlightRequest request) {
      * @return true iff we have no requests still being sent to the given node
      */
     public boolean canSendMore(String node) {
-        Deque<NetworkClient.InFlightRequest> queue = requests.get(node);
-        return queue == null || queue.isEmpty() ||
-               (queue.peekFirst().send.completed() && queue.size() < this.maxInFlightRequestsPerConnection);
+        return this.maxInFlightRequestsPerConnection - count(node) > 0;
```

Comment:
I think the existing behaviour is intentional. The idea is that we want to have at most a single `send` in progress at any given time. This is important so that we maintain ordering. However, once a `send` is complete (i.e. we have sent all the bytes) and we are waiting for a response (which can take time as the server may be waiting for replication to happen, for example), we can then send more requests if `queue.size < maxInFlightRequestsPerConnection`. Does that make sense?